packages/carousel/src/react/index.js

```diff
@@ -234,3 +258,13 @@ function usePager(pageCount, additionalSideEffectTriggers = []) {
     ref
   }
 }
+
+function usePrevious(value) {
```

Comment:
Such a weird React 16 pattern.  Feels like a shoe-horned ability of the ref feature.

Reply:
yeah. it's because there is no hook method for componentWillReceiveNewProps so this is the way to store the prev prop(or state/context/whatever)